### PR TITLE
merging feature/cosmos/spark30 branch to master

### PR DESCRIPTION
**Pull request by moderakh**
**Please before reviewing this PR read the description.**

This PR intends to merge feature/cosmos/spark30 branch to master.

**NOTE: do NOT merge this PR. we don't want merge squash because we need to maintain history. 
please sync with @moderakh**

1) Our Spark Connector closely utilizes our Cosmos DB Java V4 SDK.
2) As this a Spark Connector, it does "not" provide SDK like public APIs. The public APIs are defined through spark framework we are only implementing the spark interfaces.
3) As discussed before, the code is written in Scala similar to other Spark Connectors in the industry.
4) We are following the agreed groupId/artifactId
com.azure.cosmos.spark:azure-cosmos-spark_3-1_2-12
refresher of the prior discussion, as per industry standards scala version needs to be part of the artifactId.
As agreed, we are using `com.azure.cosmos.spark` groupId to avoid polluting `com.azure` groupId
5) As this is Spark Connector, all its dependencies are shaded as per spark common practices.
6) To be in par with latest improvements of the Cosmos DB SDK, we need to continue the development and the releases on the master branch.

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by ealsur — APPROVED** [review]

**Comment by weshaggard** [comment]
The template pipeline failures are known and being fixed in https://github.com/Azure/azure-sdk-for-java/pull/20076 so they can be ignored for this PR.

**Review by kushagraThapar — APPROVED** [review]
LGTM from engineering perspective.

**Review by JonathanGiles — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java [hunk 1]
```diff
@@ -146,6 +147,30 @@ public CosmosChangeFeedRequestOptions setMaxPrefetchPageCount(int maxPrefetchPag
...
 
+    /**
+     * Gets the quotaInfoEnabled setting for change feed request in the Azure Cosmos DB database service.
+     * quotaInfoEnabled is used to enable/disable getting quota related stats
+     *
+     * @return true if quotaInfoEnabled is enabled
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText =
+        Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public boolean isQuotaInfoEnabled() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java:158]: @FabianMeiswinkel  @kushagraThapar FYI

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/FeedResponse.java [hunk 2]
```diff
@@ -185,6 +185,24 @@ public long getCollectionSizeUsage() {
         return this.currentQuotaHeader(Constants.Quota.COLLECTION_SIZE);
     }
 
+    /**
+     * Gets the current size of the documents in a container in kilobytes from the Azure Cosmos DB service.
+     *
+     * @return The current size of a container in kilobytes.
+     */
+    public long getDocumentUsage() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/FeedResponse.java:193]: @FabianMeiswinkel  @kushagraThapar FYI

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java:158]: the class itself is beta

**Pull request merged**